lib/membrane_live/hls/file_storage.ex

```diff
@@ -57,49 +60,63 @@ defmodule MembraneLive.HLS.FileStorage do
         contents,
         _metadata,
         %{mode: :binary},
-        %__MODULE__{directory: directory}
+        %{directory: directory} = state
       ) do
-    File.write(Path.join(directory, segment_filename), contents, [:binary])
+    {File.write(Path.join(directory, segment_filename), contents, [:binary]), state}
   end
 
   @impl true
-  def store(_parent_id, name, contents, _metadata, %{mode: :text}, %__MODULE__{
-        directory: directory
-      }) do
+  def store(
+        _parent_id,
+        name,
+        contents,
+        _metadata,
+        %{mode: :text},
+        %{directory: directory} = state
+      ) do
     result = File.write(Path.join(directory, name), contents)
 
-    if String.match?(name, ~r/\.m3u8$/) do
-      notify_playlist_update(directory, name, contents)
-    end
+    state =
+      if String.match?(name, ~r/\.m3u8$/) do
+        notify_playlist_update(name, contents, state)
+      else
+        state
+      end
 
-    result
+    {result, state}
   end
 
   @impl true
-  def remove(_parent_id, name, _ctx, %__MODULE__{directory: location}) do
-    File.rm(Path.join(location, name))
+  def remove(_parent_id, name, _ctx, %{directory: location} = state) do
+    {File.rm(Path.join(location, name)), state}
   end
 
-  defp notify_playlist_update(directory, name, contents) do
-    name_without_extension = String.replace(name, ".m3u8", "")
-
-    if send_first_segment_notification?(contents) do
-      "output/" <> event_id = directory
-      PubSub.broadcast(MembraneLive.PubSub, event_id, :first_segment_ready)
-    end
+  defp notify_playlist_update(name, contents, state) do
+    state = maybe_send_first_segment_notification(state, contents)
 
     {segment, partial} = Helpers.get_last_partial(contents)
+    name_without_extension = String.replace(name, ".m3u8", "")
 
     PubSub.broadcast(
       MembraneLive.PubSub,
       name_without_extension,
       {:manifest_update_partial, segment, partial}
     )
```

Comment:
theoretically, we don't have to send partials till param `first_segment_ready` is true.

Reply:
don't forget about it.